src/spacel/provision/template/app.py

```diff
@@ -90,19 +93,29 @@ def app(self, app, region):
                 })
 
             # Allow ELB->Instance
-            instance_ingress.append({
-                'IpProtocol': 'tcp',
-                'FromPort': port_number,
-                'ToPort': port_number,
-                'SourceSecurityGroupId': {'Ref': 'ElbSg'}
-            })
+            internal_port = port_config.internal_port
+            if internal_port not in elb_ingress_ports:
+                instance_ingress.append({
+                    'IpProtocol': 'tcp',
+                    'FromPort': internal_port,
+                    'ToPort': internal_port,
+                    'SourceSecurityGroupId': {'Ref': 'ElbSg'}
+                })
+                elb_ingress_ports.add(internal_port)
 
             elb_listener = {
-                'InstancePort': port_number,
+                'InstancePort': internal_port,
                 'LoadBalancerPort': port_number,
                 'Protocol': port_config.scheme,
                 'InstanceProtocol': port_config.internal_scheme
             }
+
+            if port_config.scheme in SSL_SCHEMES:
+                cert = port_config.certificate
+                if not cert:
+                    cert = self._acm.get_certificate(region, app_hostname)
+                elb_listener['SSLCertificateId'] = cert
```

Comment:
Good point...
Right now it's relying on CF to catch that issue, I'll update to throw an exception here to fail faster.

I don't think there's a good fallback: downgrading to HTTP is.. nope, and skipping the listener seems cruel too (you need to _really_ pay attention to the log output).

Nuclear option only option?